Expensify/App
# Fix the modal dismiss behavior

### Details
Adds the ability to dismiss modals back to either the LHN or the last screen they were triggered from (so just the Report screen at the moment).

### Fixed Issues
Fixes https://github.com/Expensify/Expensify/issues/157574

### Tests
### QA Steps
1. Pull up a chat with one user and click on the avatar to open "Details"
2. Tap the `x` in the top right corner to dismiss the modal
3. Verify you are brought back to the chat and not the LHN
4. Tap to launch the "Settings" screen
5. Tap the `x` in the top right corner to dismiss the modal
6. Verify you are brought back to the LHN
7. Tap to launch the "Settings" screen
8. Tap to launch "Profile"
9. Tap the `x` in the top right corner to dismiss the modal
10. Verify you are brought back to the LHN
11. Repeat for all sub screens: **Profile, Preferences, Change Password, Payments etc.**

### Tested On


### Screenshots


#### Web
![2021-04-07_17-22-14 (1)](https://user-images.githubusercontent.com/32969087/113963819-02217d00-97c6-11eb-8b20-ceb0fc64d783.gif)

#### Mobile Web
![2021-04-07_17-24-41 (1)](https://user-images.githubusercontent.com/32969087/113963915-3137ee80-97c6-11eb-94f7-96d26d188a68.gif)

#### Desktop
![2021-04-07_17-28-31 (1)](https://user-images.githubusercontent.com/32969087/113964212-bfac7000-97c6-11eb-8cbd-849e9aae5040.gif)

#### iOS
![2021-04-07_17-33-39 (1)](https://user-images.githubusercontent.com/32969087/113964596-845e7100-97c7-11eb-9d36-92627f459c74.gif)

#### Android
![2021-04-07_17-38-37 (1)](https://user-images.githubusercontent.com/32969087/113964923-2d0cd080-97c8-11eb-80c7-a90b32f5b148.gif)
